plugins/modules/vlan_member.py

```diff
@@ -0,0 +1,136 @@
+        required: true
+        type: int
+    state:
+        description: Whether the interface should be attached to the VLAN or not
+        default: present
+        type: str
+        choices: [present, absent]
+    interface:
+        description: The interface to attach to the VLAN
+        required: true
+        type: str
+    tagging_mode:
+        description: Whether the VLAN should be tagged for the interface or not
+        default: tagged
+        type: str
+        choices: [tagged, untagged]
+
+author:
+    - Erik Larsson (@whooo)
+'''
+
+EXAMPLES = r'''
+# Add an interface to VLAN
+- name: Add interface to VLAN
+  community.sonic.vlan_member:
+    vlanid: 3600
+    interface: Ethernet90
+
+# Remove an interface from a VLAN
+- name: Remove interface from VLAN
+  community.sonic.vlan_member:
+    vlanid: 3600
+    state: absent
+    interface: Ethernet90
+
+# Add an untagged interface
+- name: VLAN membership untagged
+  community.sonic.vlan_member:
+    vlanid: 3600
+    interface: Ethernet90
+    tagging_mode: untagged
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    from swsscommon import swsscommon
+except ImportError:
+    HAS_SWSSCOMMON_LIBRARY = False
+    SWSSCOMMON_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_SWSSCOMMON_LIBRARY = True
+    SWSSCOMMON_IMPORT_ERROR = None
+
+
+def build_vlan_member_entry(vlanid, interface, state, tagging_mode):
+    key = f'Vlan{vlanid}|{interface}'
+    if state == 'absent':
+        return key, None
+
+    val = {
+        'tagging_mode': tagging_mode,
+    }
+    return key, val
+
+
+def compare_vlan_member_entries(new, cur):
+    # if both new and cur is empty/None, both current and wanted state is absent
+    if not new and not cur:
+        return False
+    # if new XOR cur is empty/None current and wanted state doesn't match
+    elif (not new) != (not cur):
+        return True
+    if new.get("tagging_mode") != cur.get("tagging_mode"):
+        return True
+    return False
+
+
+def run_module():
+    module_args = dict(
+        vlanid=dict(type='int', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent'], required=False),
+        interface=dict(type='str', required=True),
+        tagging_mode=dict(type='str', default='tagged', choices=['tagged', 'untagged'], required=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if not HAS_SWSSCOMMON_LIBRARY:
+        module.fail_json(
+            msg=missing_required_lib('swsscommon'),
+            exception=SWSSCOMMON_IMPORT_ERROR)
+
+    config_db = swsscommon.ConfigDBConnector()
+    config_db.connect()
+
+    key, val = build_vlan_member_entry(**module.params)
+    cur_val = config_db.get_entry('VLAN_MEMBER', key)
+    changed = compare_vlan_member_entries(val, cur_val)
+
+    if module.check_mode or not changed:
```

Comment:
The behavior when the existing key contains an extra property not implemented by this module is too hidden for my taste. I would like to see a test that cements how these extra properties are handled.

Like this:
```py
VLAN_TABLE = {
    'Vlan3361': {
        'vlanid': 3361,
        'extra-option': 1234,
    }
}

_, val = vlan.build_vlan_entry(vlanid=3361, state='present')
changed = vlan.compare_vlan_entries(val, VLAN_TABLE.get('Vlan3361'))
assert not changed, 'extra options should be silently ignored'

```

What do you think?

Reply:
Yeah, I missed that, doing a deep comparison of both dicts is probably better

Reply:
I reworked it by adding a utility function for comparing two (possible empty/None) entries. so any option the modules aren't aware of will still lead to changed being true